src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -24,12 +24,20 @@ class WP_HTML_Processor extends WP_HTML_Tag_Processor {
 	 * @return boolean Whether an element was found.
 	 */
 	public function step( $insertion_mode = null ) {
-		switch ( $insertion_mode ?: $this->insertion_mode ) {
-			case 'in-body':
-				return $this->step_in_body();
+		try {
+			switch ( $insertion_mode ?: $this->insertion_mode ) {
+				case 'in-body':
+					return $this->step_in_body();
 
-			default:
-				return self::NOT_IMPLEMENTED_YET;
+				default:
+					return self::NOT_IMPLEMENTED_YET;
+			}
+		} catch ( Exception $e ) {
```

Comment:
Catching all exceptions might hide legitimate errors, it would be safer to have a HTML_Exception class or so

Reply:
good point. at some point I wanted to go back and special-case known exceptions so we don't hide others, but I didn't do that yet because I wasn't entirely sure how I felt about the escaping mechanism as throwing Exceptions.